description: gcc requires these structs to be defined at namespace scope
author: Michael Gilbert <mgilbert@debian.org>

--- a/components/services/leveldb/leveldb_database_impl.cc
+++ b/components/services/leveldb/leveldb_database_impl.cc
@@ -152,12 +152,13 @@ void LevelDBDatabaseImpl::Write(
       std::move(callback));
 }
 
+struct GetResult {
+  Status status;
+  storage::DomStorageDatabase::Value value;
+};
+
 void LevelDBDatabaseImpl::Get(const std::vector<uint8_t>& key,
                               GetCallback callback) {
-  struct GetResult {
-    Status status;
-    storage::DomStorageDatabase::Value value;
-  };
   RunDatabaseTask(base::BindOnce(
                       [](const std::vector<uint8_t>& key,
                          const storage::DomStorageDatabase& db) {
@@ -173,13 +174,13 @@ void LevelDBDatabaseImpl::Get(const std:
                       std::move(callback)));
 }
 
+struct GetPrefixedResult {
+  Status status;
+  std::vector<storage::DomStorageDatabase::KeyValuePair> entries;
+};
+
 void LevelDBDatabaseImpl::GetPrefixed(const std::vector<uint8_t>& key_prefix,
                                       GetPrefixedCallback callback) {
-  struct GetPrefixedResult {
-    Status status;
-    std::vector<storage::DomStorageDatabase::KeyValuePair> entries;
-  };
-
   RunDatabaseTask(
       base::BindOnce(
           [](const std::vector<uint8_t>& prefix,
